pkgs/development/libraries/boost/1.76.nix

```diff
@@ -0,0 +1,14 @@
+{ callPackage, fetchurl, fetchpatch, ... } @ args:
+
+callPackage ./generic.nix (args // rec {
+  version = "1.76.0";
+
+  src = fetchurl {
+    urls = [
+      "mirror://sourceforge/boost/boost_1_76_0.tar.bz2"
+      "https://dl.bintray.com/boostorg/release/1.76.0/source/boost_1_76_0.tar.bz2"
+    ];
+    # SHA256 from http://www.boost.org/users/history/version_1_76_0.html
```

Comment:
Why are those version numbers not substituted?

Reply:
I think because nixpkgs carries every major version of boost starting with 1.55, the version-specific files aren't generalized since they are effectively "pinned" forever.

Reply:
Perhaps it would be better to start fetching from https://github.com/boostorg/boost/releases instead of their sourceforge mirrors? Also i'm not sure why nixpkgs "default boost" is still 1.6. I can try to bump it, but this PR will need to go to staging for that since it will be the mother of all mass rebuilds.

Reply:
> Perhaps it would be better to start fetching from boostorg/boost/releases instead of their sourceforge mirrors?

I think that is a good idea for new releases. It also makes fetching patches easier.

> Also i'm not sure why nixpkgs "default boost" is still 1.6.

Probably because many packages break if we change it.

> this PR will need to go to staging

It will be staging and that is not a problem.